### PR TITLE
TCOMP-1851 : add @Base annotation for http request

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-1851

### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
Give possibility to make Http Service stateless.

### What does this PR adds (design/code thoughts)?
Annotation @Base for services.

### CODE REVIEW

File: component-api/src/main/java/org/talend/sdk/component/api/service/http/Base.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Target(PARAMETER)
+@Retention(RUNTIME)
+public @interface Base {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/service/http/Base.java:26]: Please add a `@Documentation` annotation for the purpose of this annotation.

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/http/RequestParser.java [hunk 2]
```diff
@@ -83,9 +85,34 @@
...
+
+    @AllArgsConstructor
+    final static class ReflectionInstanceCreator implements InstanceCreator {
+
+        private final ReflectionService reflections;
+
+        private final Map<Class<?>, Object> services;
+
+        @Override
+        public <T> T buildNew(final Class<? extends T> realClass) {
```
Comment (undx) [inline component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/http/RequestParser.java:103]: strange name but overriding... so ok ;-)

**Review by undx — APPROVED** [review]
Some minor comments.
Did you test changes on existing connectors? 
Maybe add more tests like 2 connectors w/ 2 distinct base...

File: component-api/src/main/java/org/talend/sdk/component/api/service/http/Base.java (hunk 1 above)
Comment (clesaec) [reply component-api/src/main/java/org/talend/sdk/component/api/service/http/Base.java:32]: done

**Pull request merged**